### PR TITLE
Get the VisualStudio solution working

### PR DESCRIPTION
**Pull request by TravisEz13**
## PR Summary

Forking from #5209 
- Changing the project type GUID.  This should make the `TargetFramework` not required to 
  build in Visual Studio for Windows (works on Mac without the change.)
- Add the `powershell-unix` project so you can choose which one to build.
- remove a resource that wasn't used and fails in VS from two resource files.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: PowerShell-Win.sln [hunk 1]
```diff
@@ -1,33 +1,35 @@
...
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.WSMan.Management", "src\Microsoft.WSMan.Management\Microsoft.WSMan.Management.csproj", "{8F63D134-E413-4181-936D-D82F3F5F1D85}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Microsoft.PowerShell.Security", "src\Microsoft.PowerShell.Security\Microsoft.PowerShell.Security.csproj", "{C4F81816-C87A-4ABF-8A37-24AC16A0A6CF}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.PowerShell.Security", "src\Microsoft.PowerShell.Security\Microsoft.PowerShell.Security.csproj", "{C4F81816-C87A-4ABF-8A37-24AC16A0A6CF}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Microsoft.WSMan.Runtime", "src\Microsoft.WSMan.Runtime\Microsoft.WSMan.Runtime.csproj", "{D9CCCB67-4EBE-4854-AB52-C0129DC5BAE4}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.WSMan.Runtime", "src\Microsoft.WSMan.Runtime\Microsoft.WSMan.Runtime.csproj", "{D9CCCB67-4EBE-4854-AB52-C0129DC5BAE4}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "powershell-unix", "src\powershell-unix\powershell-unix.csproj", "{73EA0BE6-C0C5-4B56-A5AA-DADA4C01D690}"
```
Comment (daxian-dbw) [inline PowerShell-Win.sln]: Want to change this one as well?

File: PowerShell-Win.sln [hunk 2]
```diff
@@ -37,52 +39,52 @@ Global
...
-		{8359D422-E0C4-4A0D-94EB-3C9DD16B7932}.CodeCoverage|Any CPU.ActiveCfg = CodeCoverage|Any CPU
-		{8359D422-E0C4-4A0D-94EB-3C9DD16B7932}.CodeCoverage|Any CPU.Build.0 = CodeCoverage|Any CPU
+		{8359D422-E0C4-4A0D-94EB-3C9DD16B7932}.CodeCoverage|Any CPU.ActiveCfg = Release|Any CPU
+		{8359D422-E0C4-4A0D-94EB-3C9DD16B7932}.CodeCoverage|Any CPU.Build.0 = Release|Any CPU
 		{8359D422-E0C4-4A0D-94EB-3C9DD16B7932}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{8359D422-E0C4-4A0D-94EB-3C9DD16B7932}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{8359D422-E0C4-4A0D-94EB-3C9DD16B7932}.Linux|Any CPU.ActiveCfg = Linux|Any CPU
-		{8359D422-E0C4-4A0D-94EB-3C9DD16B7932}.Linux|Any CPU.Build.0 = Linux|Any CPU
+		{8359D422-E0C4-4A0D-94EB-3C9DD16B7932}.Linux|Any CPU.ActiveCfg = Release|Any CPU
+		{8359D422-E0C4-4A0D-94EB-3C9DD16B7932}.Linux|Any CPU.Build.0 = Release|Any CPU
```
Comment (daxian-dbw) [inline PowerShell-Win.sln]: This GUID points to `powershell-win-core`, so maybe `Linux|Any` is not needed.

File: PowerShell-Win.sln [hunk 3]
```diff
@@ -101,46 +103,48 @@ Global
...
 		{D9CCCB67-4EBE-4854-AB52-C0129DC5BAE4}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{D9CCCB67-4EBE-4854-AB52-C0129DC5BAE4}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{D9CCCB67-4EBE-4854-AB52-C0129DC5BAE4}.Linux|Any CPU.ActiveCfg = Linux|Any CPU
-		{D9CCCB67-4EBE-4854-AB52-C0129DC5BAE4}.Linux|Any CPU.Build.0 = Linux|Any CPU
+		{D9CCCB67-4EBE-4854-AB52-C0129DC5BAE4}.Linux|Any CPU.ActiveCfg = Release|Any CPU
+		{D9CCCB67-4EBE-4854-AB52-C0129DC5BAE4}.Linux|Any CPU.Build.0 = Release|Any CPU
 		{D9CCCB67-4EBE-4854-AB52-C0129DC5BAE4}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{D9CCCB67-4EBE-4854-AB52-C0129DC5BAE4}.Release|Any CPU.Build.0 = Release|Any CPU
+		{73EA0BE6-C0C5-4B56-A5AA-DADA4C01D690}.Linux|Any CPU.ActiveCfg = Linux|Any CPU
+		{73EA0BE6-C0C5-4B56-A5AA-DADA4C01D690}.Linux|Any CPU.Build.0 = Linux|Any CPU
```
Comment (daxian-dbw) [inline PowerShell-Win.sln]: > Linux|Any CPU

Should this be changed to `Release|Any CPU`?

File: PowerShell-Win.sln [hunk 4]
```diff
@@ -1,33 +1,35 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 15
```
Comment (daxian-dbw) [inline PowerShell-Win.sln]: Can we have a comment here pointing to  https://github.com/dotnet/project-system/blob/master/docs/opening-with-new-project-system.md#project-type-guids ?

File: src/Microsoft.PowerShell.Commands.Management/resources/CmdletizationResources.resx [hunk 5]
```diff
@@ -124,10 +124,6 @@
 {1} is a placeholder for a server name.  Example: "localhost".</comment>
   </data>
   <assembly alias="System.Windows.Forms" name="System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" />
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.Commands.Management/resources/CmdletizationResources.resx]: I thought this file can be removed, not?

File: PowerShell-Win.sln (hunk 2 above)
Comment (TravisEz13) [reply PowerShell-Win.sln]: The whole solution has to support Linux... so I have to map it to a valid configuration for this project

File: src/Microsoft.PowerShell.Commands.Management/resources/CmdletizationResources.resx (hunk 5 above)
Comment (TravisEz13) [reply src/Microsoft.PowerShell.Commands.Management/resources/CmdletizationResources.resx]: no, it's used and a different resource.

File: PowerShell-Win.sln [hunk 6]
```diff
@@ -3,31 +3,33 @@ Microsoft Visual Studio Solution File, Format Version 12.00
...
```
Comment (daxian-dbw) [inline PowerShell-Win.sln]: Can we have a comment here pointing to https://github.com/dotnet/project-system/blob/master/docs/opening-with-new-project-system.md#project-type-guids ?

**Review by daxian-dbw — APPROVED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**